### DISCUSSION TITLE
Override nni_alloc and nni_free

### DISCUSSION THREAD

**Discussion by sylvain-iRacing** [nanomsg/nng#1830] [Q&A] [answered]
Is there a way to override nni_alloc and nni_free?  I cannot find any hook to redirect them to the application allocators. Redefining them is not an option due to linking errors by doing so.

**Comment by gdamore** [comment, marked as answer]
Not directly.  There is a platform layer, and on POSIX they sit on top of malloc and free, so if you have weak bindings along with a shared library shim, you might be able to override that.

I'd be willing to consider a pluggable allocator.

**Reply by GavinLaz** [reply]
My project could really benefit from being able to plug in a custom allocator.  I was thinking about trying to submit a PR for doing that implementation but I am still quite new to the details of NNG.  I may eventually get there….but in the mean time if you’re at all considering doing this, consider this a giant PLUS ONE request (although given how old this question is….maybe I’m the only one currently asking for it. 🤣).
